src/interfaces/coral_web/src/components/Shared/DragDropFileInput/DragDropFileInput.tsx

```diff
@@ -93,11 +96,52 @@ export const DragDropFileInput: React.FC<DragDropFileInputProps> = ({
         placeholder={placeholder}
         disabled={disabled}
         readOnly={readOnly}
-        onChange={onChange}
         onDragEnter={() => setDragActive(true)}
         onDragOver={() => setDragActive(true)}
         onDragLeave={() => setDragActive(false)}
-        onDrop={() => setDragActive(false)}
+        onDrop={async (e) => {
+          const droppedFilesSystem = e.dataTransfer.items[0].webkitGetAsEntry();
+          if (!droppedFilesSystem) return;
+          const filesList: File[] = [];
+
+          const traverseFolder = async (fileSystem: FileSystemDirectoryEntry): Promise<void> => {
+            if (fileSystem.isDirectory) {
+              const directoryReader = fileSystem.createReader();
+              return new Promise((resolve) => {
+                directoryReader.readEntries(async function (entries) {
+                  for (let i = 0; i < entries.length; i++) {
+                    const entry = entries[i];
+                    console.debug(entry);
+
+                    if (IGNORED_FILES.includes(entry.name)) {
+                      continue;
+                    }
+                    if (entry.isDirectory) {
+                      await traverseFolder(entry as FileSystemDirectoryEntry);
+                    } else if (entry.isFile) {
+                      const fileEntry = entry as FileSystemFileEntry;
+                      const readFile = () =>
+                        new Promise((fileReadResolve) => {
+                          fileEntry.file((f) => {
+                            filesList.push(f);
+                            fileReadResolve(f);
+                          });
+                        });
+                      await readFile();
+                    }
+                    if (i === entries.length - 1) {
+                      resolve();
```

Comment:
`readEntries` doesn't return a promise so can't be awaited 😅 this is my work around so that the `filesList` can be fully populated before sending it off to be uploaded